src/ai/susi/mind/SusiSkill.java

```diff
@@ -511,6 +518,7 @@ public static JSONObject getSkillMetadata(String model, String group, String lan
                     skillid.hasName(skillname)) {
 
                 skillMetadata.put("skill_name", skill.getSkillName() ==null ? JSONObject.NULL: skill.getSkillName());
+                skillMetadata.put("private_skill", skill.getPrivateSkill());
```

Comment:
Can you try adding null check here as done for other parameters?

Reply:
It's a `boolean` with initial value `false`. Null checks are for strings.